### PR TITLE
docs(path): explain path library behavior.

### PR DESCRIPTION
**Pull request by illusory0x0**
test(path): add test for edge cases.

@tonyfettes  This PR should review more carefully, path library has many confused behavior.

### CODE REVIEW

File: internal/path/path.mbt [hunk 1]
```diff
@@ -158,15 +179,20 @@ pub fn normalize(view : StringView) -> String {
 }
 
 ///|
+/// 1. normalize path.
+/// 2. if rhs is absolute path, return rhs.
 pub fn join(p : StringView, q : StringView) -> String {
   let p = normalize(p)
   let q = normalize(q)
+  // confuse behvior.
+  // Python' os.path.join doesn't has this behavior.
```
[the remark this answers is not in the document]
Comment (illusory0x0) [reply internal/path/path.mbt]: Note: 

```python
>>> import os.path as path
>>> path.join("a",".")
'a/.'
```

yields on macOS.

File: internal/path/path.mbt [hunk 2]
```diff
@@ -188,6 +214,8 @@ pub fn is_relative(path : StringView) -> Bool {
 }
 
 ///|
+/// Return relative path from `from` to `to`, `from` treated as current working directory.
+///
```
[the remark this answers is not in the document]
Comment (illusory0x0) [reply internal/path/path.mbt]: > ```diff
> - normalize(join(from, relative(from, to))) == to
> + join(from, relative(from, to)) == resolve(to)
> ```

join already normalize, `to` should resolve.

**Comment by illusory0x0** [comment]
cc @tonyfettes I think all conversation solved, do you approve this PR?

**Review by tonyfettes — APPROVED** [review]
